### PR TITLE
Introduce rubocop

### PR DESCRIPTION
**Pull request by treiff**
I added the `.rubocop.yml` from the codeunion repo.  I had to modify it slightly, here's a quick summary of what I changed:
* Require double-quotes
* Prefer new hash syntax
* No braces around hash parameters
* Exclude files from `db/schema` and `db/migrate`
* Exclude documentation cops
  * prevent top-level class documentation errors
* Exclude block comments
* Exclude useless assignments to variables
  * `user_spec.rb` throwing error due to `user1` not being called.

All cops passing now.  I was originally committing by file, but later started committing by directory to avoid tons of commits. 

Let me know if you need me to look over something else with this.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+AllCops:
+  Exclude:
+    - '*.gemspec'
+    - 'bin/*'
+    - 'db/migrate/*'
+    - 'db/schema.rb'
+
+Style/Documentation:
+  Enabled: false
```
Comment (zspencer) [inline .rubocop.yml]: We actually *want* to be documenting our classes.

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  Max: 20
+
+Metrics/LineLength:
+  Max: 120
+
+Metrics/AbcSize:
+  Max: 25
+
+Lint/UselessAssignment:
+  Enabled: false
```
Comment (zspencer) [inline .rubocop.yml]: We actually want this to be enabled as well. Better to fix the implementation that has additional, useless variables.

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - '*.gemspec'
+    - 'bin/*'
+    - 'db/migrate/*'
+    - 'db/schema.rb'
+
+Style/Documentation:
+  Enabled: false
+
+Style/LineEndConcatenation:
+  Enabled: false
```
Comment (zspencer) [inline .rubocop.yml]: Why are we disabling this?

File: .rubocop.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+Style/StringLiteralsInInterpolation:
+  Enabled: true
+  EnforcedStyle: double_quotes
+
+Style/MethodCallParentheses:
+  Enabled: false
+
+Metrics/MethodLength:
+  Max: 20
```
Comment (zspencer) [inline .rubocop.yml]: This seems really long, why did we pick that?

**Comment by zspencer** [comment]
I  prefer to require class-level comments. It ensure's we're defining the "One Job" of any given class or module.

File: .rubocop.yml (hunk 3 above)
Comment (treiff) [inline .rubocop.yml]: I assume in these comments were talking about lines 11-12.  If so, this was carried over from when I copied it over from codeunion.  I'll re-enable.

Comment (zspencer) [inline .rubocop.yml]: Nah that's a valid reason. I wasn't sure why it was enabled. "Doing what CodeUnion Does" is OK.

File: .rubocop.yml (hunk 4 above)
Comment (treiff) [inline .rubocop.yml]: Same with this, left over.  What's a recommended max method name length?

**Comment by treiff** [comment]
Ok, I'm going to grab a bite then re-visit this.  
* Class level comments.
* Fix the spec implementation with unused variables.

**Comment by zspencer** [comment]
:+1: Looks good man!

File: .rubocop.yml (hunk 4 above)
Comment (zspencer) [inline .rubocop.yml]: I believe [MethodLength](http://www.rubydoc.info/github/bbatsov/rubocop/master/Rubocop/Cop/Style/MethodLength) is about lines-of-code, not length of method name.

I would probably leave this on the default until we run into problems. 20 line methods are pretty big!

Comment (zspencer) [inline .rubocop.yml]: I think the reason it was set to 20 in the client is we wanted to make it easier for jr. devs to jump in, whiich often means limiting the amount of abstraction.

**Pull request merged**